### PR TITLE
Update documentation on retention policies limits

### PR DESCRIPTION
**Pull request by babolivier**
Document the changes from https://github.com/matrix-org/synapse/pull/8104

### CODE REVIEW

File: docs/message_retention_policies.md [hunk 1]
```diff
@@ -136,24 +136,34 @@ the server's database.
...
   allowed_lifetime_min: 1d
   allowed_lifetime_max: 1y
 ```
 
-Here, `allowed_lifetime_min` is the lowest value a local user can set
-for both `min_lifetime` and `max_lifetime`, and `allowed_lifetime_max`
-is the highest value. Both parameters are optional (e.g. setting
-`allowed_lifetime_min` but not `allowed_lifetime_max` only enforces a
-minimum and no maximum).
+These values are evaluated when running purge jobs, to adjust the
```
Comment (anoadragon453) [inline docs/message_retention_policies.md:148]:
```suggestion
The limits are considered when running purge jobs, to adjust the
```

File: docs/message_retention_policies.md [hunk 2]
```diff
@@ -136,24 +136,34 @@ the server's database.
...
+effective value of `max_lifetime`, and make sure it's within the
```
Comment (anoadragon453) [inline docs/message_retention_policies.md:149]:
```suggestion
effective value of `max_lifetime`, to make sure it's within the
```

File: docs/message_retention_policies.md [hunk 3]
```diff
@@ -136,24 +136,34 @@ the server's database.
...
+`[allowed_lifetime_min, allowed_lifetime_max]` range. If the value of
```
Comment (anoadragon453) [inline docs/message_retention_policies.md:150]:
```suggestion
`[allowed_lifetime_min, allowed_lifetime_max]` range, inclusive. If the value of
```

**Review by anoadragon453 — APPROVED** [review]
Some small things, otherwise lgtm

File: docs/message_retention_policies.md (hunk 3 above)
Comment (babolivier) [reply docs/message_retention_policies.md:150]: The writing `[a, b]` already imply it's inclusive. An exclusive version would look like `]a, b[` (fully exclusive) or `[a, b[` (partially exclusive). See https://en.wikipedia.org/wiki/Interval_(mathematics)#Including_or_excluding_endpoints

File: docs/message_retention_policies.md (hunk 2 above)
Comment (babolivier) [reply docs/message_retention_policies.md:149]: Not sure about the "..., to ..., to ..." formulation, it looks more hard to read to me.

File: docs/message_retention_policies.md (hunk 3 above)
Comment (anoadragon453) [reply docs/message_retention_policies.md:150]: I know, but server admins may not understand mathematical notation is all.

File: docs/message_retention_policies.md (hunk 2 above)
Comment (anoadragon453) [reply docs/message_retention_policies.md:149]: Sorry, yes that doesn't make sense. I think this sentence is a bit of a run-on though. I'd split it up with:

The limits are considered when running purge jobs. If necessary, the value of `max_lifetime` will be brought within the `[allowed_lifetime_min, allowed_lifetime_max]` range.

File: docs/message_retention_policies.md (hunk 3 above)
Comment (babolivier) [reply docs/message_retention_policies.md:150]: I'll probably rephrase it so it uses plain English instead of a weird mix of both then.

Comment (babolivier) [reply docs/message_retention_policies.md]: Done in https://github.com/matrix-org/synapse/pull/8529/commits/ddd1f2bc9af1130e5ce209eb44f4c3c168115f19

File: docs/message_retention_policies.md (hunk 2 above)
Comment (babolivier) [reply docs/message_retention_policies.md]: Fixed in https://github.com/matrix-org/synapse/pull/8529/commits/ddd1f2bc9af1130e5ce209eb44f4c3c168115f19

**Comment by babolivier** [comment]
@anoadragon453 approved the changes in #synapse-dev so I'm going to go ahead and merge it.

**Pull request merged**